WordPress/src/main/java/org/wordpress/android/ui/posts/prepublishing/home/viewmodel/slice/SyncPublishingUiHandler.kt

```diff
@@ -0,0 +1,229 @@
+package org.wordpress.android.ui.posts.prepublishing.home.viewmodel.slice
+
+import android.util.Log
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import org.greenrobot.eventbus.EventBus
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.model.MediaModel
+import org.wordpress.android.fluxc.model.PostImmutableModel
+import org.wordpress.android.fluxc.model.PostModel
+import org.wordpress.android.fluxc.store.MediaStore
+import org.wordpress.android.fluxc.store.PostStore
+import org.wordpress.android.ui.posts.prepublishing.home.PrepublishingHomeItemUiState
+import org.wordpress.android.ui.posts.prepublishing.publishing.PublishingEvent
+import org.wordpress.android.ui.uploads.PostEvents
+import org.wordpress.android.ui.uploads.ProgressEvent
+import org.wordpress.android.ui.uploads.UploadService
+
+class SyncPublishingUiHandler(private val dispatcher: Dispatcher)
```

Comment:
This is replacing the PublishingViewModel. I added the event bus listeners in here so we could receive the uploading/publishing events directly and not through the EditPostActivity. This is a candidate components to place the code of @AjeshRPai current task.